web/modules/custom/asset_release/src/Controller/AssetReleaseController.php

```diff
@@ -73,10 +99,7 @@ public function add(RdfInterface $rdf_entity) {
    *   The access result object.
    */
   public function createAssetReleaseAccess(RdfInterface $rdf_entity) {
-    $user = $this->currentUser();
-    $membership = Og::getMembership($user, $rdf_entity);
-    // @todo: Remove check for empty after ISAICP-2369 is in.
-    return (!empty($membership) && $membership->hasPermission('create asset_release rdf_entity')) ? AccessResult::allowed() : AccessResult::forbidden();
+    return $this->ogAccess->userAccessEntity('create', $this->createNewAssetRelease($rdf_entity), $this->currentUser());
```

Comment:
This is how it is intended to be done in OG at the moment. OG doesn't differentiate between the create operation and operations on existing group content entities, such as 'edit' and 'delete'. 

The reason for this is that `hook_entity_create_access()` is not usable for group content in OG since it doesn't pass the group context. The standard routes that Drupal uses for creating group content (like `node/add`) don't contain any information about the group to which the group content belongs. We need to use custom routes for this (like `rdf_entity/{rdf_entity}/asset_release/add`) and hence a custom access check. In the future similar routes will be added to `og_ui` but we won't use that in this project.

Inside OG the access checking on CRUD operations for group content is handled by `OgAccess::userAccessGroupContentEntityOperations()` which accepts only a populated group content entity like @idimopoulos is doing here.

Note that you cannot use `OgAccess::userAccess()` for this. This is intended to check a user's access to operations on groups. It doesn't work with group content.

You _can_ use `OgAccess::userAccessEntity()` (as is currently done here) but it's a waste of cycles. This is simply a wrapper that is an entry point for `hook_entity_access()`. It does some checks on the entity to determine whether it is a group or group content, and then delegates to either `OgAccess::userAccess()` or `OgAccess::userAccessGroupContentEntityOperations()` .

I do like the suggestion that it should not be necessary for us as the API consumers to create this entity manually. OG should provide a convenience method `OgAccess::userAccessGroupContentEntityCreate(EntityInterface $group, string $group_content_type, string $group_content_bundle)` that can give us the desired result.

@brummbar if you want you can create an issue in https://github.com/amitaibu/og/issues to propose this (or I can do it for you).

For this particular hunk, I would suggest to keep the current approach but change `$this->ogAccess->userAccessEntity()` to `$this->ogAccess->userAccessGroupContentEntityOperations()`.